net/mwan3/files/usr/sbin/mwan3track

```diff
@@ -141,7 +141,7 @@ main() {
 						# https://bugs.openwrt.org/index.php?do=details&task_id=2897
 						# so get the IP address of the interface and use that instead
 						if echo $track_ip | grep -q ':'; then
-							ADDR=$(ip -6 addr ls dev "$DEVICE" | sed -ne '/\/128/d' -e 's/ *inet6 \([^ \/]*\).* scope global.*/\1/p')
+							ADDR=$(ip -6 addr ls dev "$DEVICE" scope global | sed -ne '/\/128/d' -e 's/ *inet6 \([^ \/]*\).* scope global.*/\1/p')
```

Comment:
I don't see how this can hurt, but I'm not sure it adds anything because the `sed` already only matches addresses that have `scope global` in them.

Reply:
Do we need then the filter `scope global`  in the sed command?

Reply:
It's either redundant or it's a belt-and-suspenders.  Depends on how you look at that sort of thing.  :smile:

Ultimately you could get rid of it and reduce the `sed` _expression_ down to:

```
ADDR=$(ip -6 addr ls dev "$DEVICE" scope global | sed -ne '/\/128/d' -e 's/ *inet6 \([^ \/]*\).*/\1/{p;q}')
```
(including the `{p;q}` change suggested below to only use the first address without the need for piping to `head -1`)